### ISSUE TITLE
strapi.config.get('server.url', 'fallbackUrl') returns empty string if server url is not defined

### ISSUE THREAD

**Issue by antokhio** [strapi/strapi#21987]
## Bug report
`strapi.config.get('server.url', 'fallbackUrl')` returns empty string if server url is not defined

### Required System information

- Node.js version: 18.18.2
- NPM version: 8.13.1
- Strapi version: 4.24.3
- Database: SQLite
- Operating system: Windows
- Is your project Javascript or Typescript: Typescript

### Describe the bug

Just came across an `issue` when i needed to provide a full image url to thirdparty api.
So if i have:
```ts
export default ({ env }) => ({
  host: env("HOST", "0.0.0.0"),
  port: env.int("PORT", 1337),
  app: {
    keys: env.array("APP_KEYS"),
  },
});
```
In my `config/server.ts` i expect:
```ts
const url = strapi.config.get("server.url", "http://localhost:1337");
```
to be `http://localhost:1337` 
however it's going to be an empty string

### Steps to reproduce the behavior
Testing issue is quite easy:

1. Create new strapi project
2. Run `yarn strapi console`
3. Paste command: `strapi.config.get("server.url", "http://localhost:1337")`
4. See `''` as output

### Expected behavior
It would be nice if  `strapi.config.get("server.url", "http://localhost:1337")` would return `'http://localhost:1337'`

### Additional context
You can see this behaviour cleanly when running `strapi.config.get("server")` from the console:
```
{
  host: '0.0.0.0',
  port: 1337,
  proxy: false,
  admin: { autoOpen: false },
  dirs: { public: './public' },
  app: { keys: [ 'toBeModified1', 'toBeModified2' ] },
  url: ''
}
```

This is for V4 but i'm pretty sure V5 is also affected

**Comment by kaioduarte** [comment]
That seems to be expected, the fallback value is only used if the value is undefined.

It's stated in the example in the [docs](https://docs.strapi.io/dev-docs/configurations/guides/access-configuration-values):
```js
strapi.config.get('server.host', 'defaultValueIfUndefined');

```

**Comment by antokhio** [comment]
Yes I totally understand the behaviour, but the question is why `server.url` is defined, and if it's defined, why it's not a `http://localhost:1337` in this case. I mean if you don't have url configured, you prolly expect it to be undefined or point to current url.
